dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -37,148 +43,160 @@
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				}
-				finally {
-				    try {
-		                HibernateUtil.closeSession();
-		            } catch (DotHibernateException e) {
-		                Logger.error(this, "error close session",e);
-		            }
-				}
-    		}
-    	} catch (IOException e) {
-    		Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-    	}
-
-		return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).build();
+			@QueryParam("type")        final String type,
+			@QueryParam("callback")    final String callback,
+			@QueryParam("FORCE_PUSH")  final boolean forcePush,
+			@Context final HttpServletRequest  request,
+			@Context final HttpServletResponse response
+	) throws Exception {
+		final ResourceResponse responseResource = new ResourceResponse(
+				CollectionsUtils.map("type", type, "callback", callback));
+		final String remoteIP = UtilMethods.isSet(request.getRemoteHost())?
+				request.getRemoteHost() : request.getRemoteAddr();
+
+
+		if (request.getInputStream().isFinished()) {
+			Logger.error(this.getClass(), "Push Publishing failed from " + remoteIP + " bundle expected");
+			return responseResource.responseError(HttpStatus.SC_BAD_REQUEST);
+		}
+
+		try {
+			final boolean isTokenValid = AuthCredentialPushPublishUtil.INSTANCE.processAuthHeader(request);
+
+			if (!isTokenValid) {
+				Logger.error(this.getClass(), "Invalid token from " + remoteIP + " not permission");
+				return responseResource.responseAuthenticateError("invalid_token",
+						AuthCredentialPushPublishUtil.INVALID_TOKEN_ERROR_KEY);
+			}
+
+			final Bundle bundle = this.publishBundle(forcePush, request, remoteIP);
+
+			return Response.ok(bundle).build();
+		} catch (DotSecurityException e) {
+			Logger.error(this.getClass(), "Not Admin user " + remoteIP + " not permission");
+			return responseResource.responseUnauthorizedError("admin_scope");
+		} catch(IncorrectClaimException e){
+			final String claimName = e.getClaimName();
+
+			if (Claims.EXPIRATION.equals(claimName)) {
+				return responseResource.responseAuthenticateError("invalid_token",
+						AuthCredentialPushPublishUtil.EXPIRED_TOKEN_ERROR_KEY);
+			} else {
+				return responseResource.responseAuthenticateError("invalid_token",
+						AuthCredentialPushPublishUtil.INVALID_TOKEN_ERROR_KEY);
+			}
+		}
 	}
 
-    /**
-     * Validates a received token
-     *
-     * @param token    Token to validate
-     * @param remoteIP Sender IP
-     * @param mySelf   Current end point
-     * @return True if valid
-     * @throws IOException If fails reading the security token
-     */
-    public static boolean isValidToken ( String token, String remoteIP, PublishingEndPoint mySelf ) throws IOException {
+	@WrapInTransaction
+	private Bundle publishBundle(final boolean forcePush,
+								 final HttpServletRequest request,
+								 final String remoteIP) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)